opstack/src/spec.rs

```diff
@@ -133,162 +139,174 @@ impl NetworkSpec for OpStack {
+        )
     }
 
     fn set_max_priority_fee_per_gas(&mut self, max_priority_fee_per_gas: u128) {
-        self.max_priority_fee_per_gas = Some(max_priority_fee_per_gas);
-    }
-
-    fn max_fee_per_blob_gas(&self) -> Option<u128> {
-        self.max_fee_per_blob_gas
-    }
-
-    fn set_max_fee_per_blob_gas(&mut self, max_fee_per_blob_gas: u128) {
-        self.max_fee_per_blob_gas = Some(max_fee_per_blob_gas)
+        <TransactionRequest as TransactionBuilder<Ethereum>>::set_max_priority_fee_per_gas(
+            self.as_mut(),
+            max_priority_fee_per_gas,
+        );
     }
 
-    fn gas_limit(&self) -> Option<u128> {
-        self.gas
+    fn gas_limit(&self) -> Option<u64> {
+        <TransactionRequest as TransactionBuilder<Ethereum>>::gas_limit(self.as_ref())
     }
 
-    fn set_gas_limit(&mut self, gas_limit: u128) {
-        self.gas = Some(gas_limit);
+    fn set_gas_limit(&mut self, gas_limit: u64) {
+        <TransactionRequest as TransactionBuilder<Ethereum>>::set_gas_limit(
+            self.as_mut(),
+            gas_limit,
+        );
     }
 
     fn access_list(&self) -> Option<&AccessList> {
-        self.access_list.as_ref()
+        <TransactionRequest as TransactionBuilder<Ethereum>>::access_list(self.as_ref())
     }
 
     fn set_access_list(&mut self, access_list: AccessList) {
-        self.access_list = Some(access_list);
-    }
-
-    fn blob_sidecar(&self) -> Option<&BlobTransactionSidecar> {
-        self.sidecar.as_ref()
-    }
-
-    fn set_blob_sidecar(&mut self, sidecar: BlobTransactionSidecar) {
-        TransactionBuilder::<Ethereum>::set_blob_sidecar(self, sidecar)
+        <TransactionRequest as TransactionBuilder<Ethereum>>::set_access_list(
+            self.as_mut(),
+            access_list,
+        );
     }
 
     fn complete_type(&self, ty: OpTxType) -> Result<(), Vec<&'static str>> {
         match ty {
             OpTxType::Deposit => Err(vec!["not implemented for deposit tx"]),
             _ => {
                 let ty = TxType::try_from(ty as u8).unwrap();
-                TransactionBuilder::<Ethereum>::complete_type(self, ty)
+                <TransactionRequest as TransactionBuilder<Ethereum>>::complete_type(
+                    self.as_ref(),
+                    ty,
+                )
             }
         }
     }
 
     fn can_submit(&self) -> bool {
-        TransactionBuilder::<Ethereum>::can_submit(self)
+        <TransactionRequest as TransactionBuilder<Ethereum>>::can_submit(self.as_ref())
     }
 
     fn can_build(&self) -> bool {
-        TransactionBuilder::<Ethereum>::can_build(self)
+        <TransactionRequest as TransactionBuilder<Ethereum>>::can_build(self.as_ref())
     }
 
     #[doc(alias = "output_transaction_type")]
     fn output_tx_type(&self) -> OpTxType {
-        OpTxType::try_from(self.preferred_type() as u8).unwrap()
+        match self.as_ref().preferred_type() {
+            TxType::Eip1559 | TxType::Eip4844 => OpTxType::Eip1559,
```

Comment:
Is the thought here that we should never see a EIP-4844 type here so we can just map it to 1559 to keep the function infallible?

Reply:
Yes. But full disclosure: I stole the implementation for `TransactionBuilder` from [this PR in op-alloy](https://github.com/alloy-rs/op-alloy/pull/215). So I can't say I'm sure that converting without fail is the best way for us, but that's the way alloy team decided to go.